ee_tests/src/specs/quickstart_cheterminal.spec.ts

```diff
@@ -0,0 +1,117 @@
+    let quickstart = new Quickstart(browser.params.quickstart.name);
+    let spaceName = support.newSpaceName();
+    globalSpaceName = spaceName;
+    let spaceDashboardPage = await dashboardPage.createNewSpace(spaceName);
+
+    let wizard = await spaceDashboardPage.addToSpace();
+
+    support.info('Creating quickstart: ' + quickstart.name);
+    await wizard.newQuickstartProject({ project: quickstart.name });
+    await spaceDashboardPage.ready();
+
+    /* This statement does not reliably wait for the modal dialog to disappear:
+       await browser.wait(until.not(until.visibilityOf(spaceDashboardPage.modalFade)), support.LONGEST_WAIT);
+
+       The above statement fails with this error: Failed: unknown error: Element <a id="spacehome-pipelines-title"
+       href="/username/spaceName/create/pipelines">...</a> is not clickable at point (725, 667). Other element would
+       receive the click: <modal-container class="modal fade" role="dialog" tabindex="-1" style="display:
+       block;">...</modal-container>
+
+       The only reliable way to avoid this is a sleep statement: await browser.sleep(5000); */
+    await browser.sleep(5000);
+
+    /* Open Che display page */
+    await spaceDashboardPage.codebasesSectionTitle.clickWhenReady();
+
+    let spaceChePage = new SpaceChePage();
+    await spaceChePage.createCodebase.clickWhenReady(support.LONGEST_WAIT);
+
+    /* A new browser window is opened when Che opens - switch to that new window now */
+    let handles = await browser.getAllWindowHandles();
+    support.debug('Number of browser tabs before = ' + handles.length);
+
+    handles = await browser.getAllWindowHandles();
+    while (handles.length < 2) {
+        support.info ('***Waiting for Che browser window to open***');
+        await browser.sleep (3000);
+        handles = await browser.getAllWindowHandles();
+    }
+
+    support.debug('Number of browser tabs after = ' + handles.length);
+    support.writeScreenshot('target/screenshots/che_workspace_parta_' + spaceName + '.png');
+
+    /* Switch to the Che browser window */
+    await browser.switchTo().window(handles[1]);
+
+    let spaceCheWorkSpacePage = new SpaceCheWorkspacePage();
+    support.writeScreenshot('target/screenshots/che_workspace_partb_' + spaceName + '.png');
+
+    let projectInCheTree = new Button(spaceCheWorkSpacePage.recentProjectRootByName(spaceName), 'Project in Che Tree');
+    await projectInCheTree.untilPresent(support.LONGEST_WAIT);
+    support.writeScreenshot('target/screenshots/che_workspace_partc_' + spaceName + '.png');
+
+    expect(await spaceCheWorkSpacePage.recentProjectRootByName(spaceName).getText()).toContain(spaceName);
+
+    /* Open the terminal window and execute a command */
+    await spaceCheWorkSpacePage.bottomPanelTerminalTab.clickWhenReady();
+    await support.printTerminal(spaceCheWorkSpacePage, 'cd ' + spaceName);
+
+    await support.printTerminal(spaceCheWorkSpacePage, 'mvn clean install -Popenshift,openshift-it');
```

Comment:
And this will most likely not work out of the box yet, because we are missing https://github.com/redhat-developer/rh-che/issues/541.

Pretty much what I've described [here](https://groups.google.com/d/msg/fabric8-launcher/NypXPuyoCrg/iscd40_SCgAJ)